oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -262,7 +382,15 @@ impl Runtime {
 
     /// Write a message to a channel. Fails with [`OakStatus::ErrChannelClosed`] if the underlying
     /// channel has been orphaned.
-    pub fn channel_write(&self, reference: Handle, msg: Message) -> Result<(), OakStatus> {
+    pub fn channel_write(
+        &self,
+        node_id: NodeId,
+        reference: Handle,
+        msg: Message,
+    ) -> Result<(), OakStatus> {
+        let reference = self
+            .filter_handle(node_id, reference)
+            .ok_or(OakStatus::ErrBadHandle)?;
```

Comment:
I think this would read better with the `bool` function I suggested above:

```rust
if !self.can_access_handle(node_id, reference) {
  return Err(ErrBadHandle);
}
...
```

WDYT?